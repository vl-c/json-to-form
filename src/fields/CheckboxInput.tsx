import * as React from 'react';

export interface CheckboxInputProps {
  id: string;
  label?: string;
  disabled?: boolean;
  readonly?: boolean;
  value?: boolean;
}

const CheckboxInput = (props: CheckboxInputProps): JSX.Element => {
  const { id, label, value, readonly, disabled } = props;

  return (
    <div className="input-block">
      <input
        className="input-checkbox"
        id={id}
        name={id}
        type="checkbox"
        readOnly={readonly}
        disabled={disabled}
        defaultChecked={Boolean(value)}
      />
      {label && <label htmlFor={id}>{label}</label>}
    </div>
  );
};

export default CheckboxInput;
