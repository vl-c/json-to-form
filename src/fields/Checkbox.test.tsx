import * as React from 'react';

import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CheckboxInput from './CheckboxInput';

function renderCheckboxInput() {
  return shallow(
    <CheckboxInput id={'test'} label={'CheckboxInput'} value={true} />
  );
}

it('CheckboxInput shallow render', () => {
  const res = renderCheckboxInput();
  expect(shallowToJson(res)).toMatchSnapshot();
});
