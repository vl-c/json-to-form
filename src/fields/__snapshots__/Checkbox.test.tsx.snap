// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxInput shallow render 1`] = `
<div
  className="input-block"
>
  <input
    className="input-checkbox"
    defaultChecked={true}
    id="test"
    name="test"
    type="checkbox"
  />
  <label
    htmlFor="test"
  >
    CheckboxInput
  </label>
</div>
`;
